srv/wordpress/wp-content/plugins/acore-wp-plugin/src/components/UserPanel/User.controller.php

```diff
@@ -22,56 +22,78 @@ public function showRafProgress() {
 
             if ($maxRecruitDatetime < (new \DateTime())) {
-                ?>
-                <div class="notice notice-error">
-                    <p>You can't be recruited by a friend, the 7 days limit has passed.</p>
-                </div>
-                <?php
+                $errorMessages[] = "You can't be recruited by a friend, the 7 days limit has passed.";
             } else {
                 if (!isset($_POST["recruited"])) {
-                    $errorMessage = "No recruiter value sent";
-                }
-
-                $recruiterCode = $_POST["recruited"];
-                $existingRecruiterId = $acServices->getUserNameByUserId($recruiterCode);
-                $newRecruitId = $acServices->getAcoreAccountId();
-
-                if (!$newRecruitId || !$existingRecruiterId) {
-                    $errorMessage = "Recruiter id or user id not found, please try again or contact a staff member.";
-                }
+                    $errorMessages[] = "No recruiter value sent.";
+                } else {
+                    $recruiterCode = $_POST["recruited"];
+                    $existingRecruiterId = $acServices->getUserNameByUserId($recruiterCode);
+                    $newRecruitId = $acServices->getAcoreAccountId();
 
-                if ($recruiterCode == $newRecruitId) {
-                    $errorMessage = "You can't recruit yourself.";
-                }
+                    if (!$newRecruitId || !$existingRecruiterId) {
+                        $errorMessages[] = "Recruiter id or user id not found, please try again or contact a staff member.";
+                    }
 
-                if ($errorMessage) {
-                    ?><div class="notice notice-error"><p><?= $errorMessage ?></p></div><?php
-                } else {
-                    $soap = ACoreServices::I()->getServerSoap();
-                    $res = $soap->serverInfo();
+                    if ($recruiterCode == $newRecruitId) {
+                        $errorMessages[] = "You can't recruit yourself.";
+                    }
 
-                    if ($res instanceof \Exception) {
-                        wp_die('<div class="notice notice-error"><p>Sorry, the server seems to be offline, try again later!</p></div>');
+                    if (Opts::I()->eluna_raf_config['check_ip'] === '1') {
+                        $userIp = $acServices->getAcoreAccountLastIp();
+                        $activeUserIp = "";
+                        if ( ! empty( $_SERVER['HTTP_CLIENT_IP'] ) ) {
+                        //check ip from share internet
+                        $activeUserIp = $_SERVER['HTTP_CLIENT_IP'];
+                        } elseif ( ! empty( $_SERVER['HTTP_X_FORWARDED_FOR'] ) ) {
+                        //to check ip is pass from proxy
+                        $activeUserIp = $_SERVER['HTTP_X_FORWARDED_FOR'];
+                        } else {
+                        $activeUserIp = $_SERVER['REMOTE_ADDR'];
+                        }

```

Comment:
nice, may you could improve the indentation
I am wondering what IDE you use, may you mixed tabs and spaces and you don't see the indentation like github does. Are you using atom?

Reply:
That's weird, i'm using vscode and all looked good, will take a look again

Reply:
convert tabs into space from the settings